File: app/src/main/java/com/arkapp/partyplanner/data/room/CatererDao.java
package com.arkapp.partyplanner.data.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.arkapp.partyplanner.data.models.Caterer;

import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Created by Abdul Rehman on 17-05-2020.
 * Contact email - abdulrehman0796@gmail.com
 */


/**
 * This class is used to define query for the Caterer table
 */
@Dao
public interface CatererDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(@NotNull Caterer var2);

    @Query("SELECT * FROM CATERER")
    List<Caterer> getAllCaterers();

    @Query("SELECT * FROM CATERER WHERE price*:guestCount<=:budget")
    List<Caterer> getAllCaterersInBudget(double budget, int guestCount);
}

File: app/src/main/java/com/arkapp/partyplanner/data/room/CatererDao.kt
package com.arkapp.partyplanner.data.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.arkapp.partyplanner.data.models.Caterer

/**
 * Created by Abdul Rehman on 17-05-2020.
 * Contact email - abdulrehman0796@gmail.com
 */

@Dao
interface CatererDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(vararg caterer: Caterer)

    @Query("SELECT * FROM CATERER")
    suspend fun getAllCaterers(): List<Caterer>

    @Query("SELECT * FROM CATERER WHERE price*:guestCount<=:budget")
    suspend fun getAllCaterersInBudget(budget: Double, guestCount: Int): List<Caterer>
}